artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -234,10 +280,32 @@ async def get_latest_build(self, name: str, group: str, outcome: KonfluxBuildOut
                 continue
 
         # If we got here, no builds have been found in the whole 36 months period
+        if strict:
+            raise ValueError(f"Build record for {name} not found.")
         self.logger.warning('No builds found for %s in %s with status %s in assembly %s and target %s',
                             name, group, outcome.value, assembly, el_target)
         return None
 
+    async def get_build_record_by_nvr(self, nvr: str, outcome: KonfluxBuildOutcome = KonfluxBuildOutcome.SUCCESS, strict: bool = True) -> typing.Optional[KonfluxRecord]:
```

Comment:
Once we unify the `search_builds_by_fields()` implementation, this function won't be needed as you can just search with no start_time param

Reply:
It seems to me all `get_build_record_by_nvr()` does is call `search_builds_by_fields` and raise an IOError if `strict` is True. I would rather avoid defining extra functions to search for specific fields, because if we follow with this approach more function could be defined in the future, and their implementation would eventually diverge leaving us with a codebase that's hard to maintain and keep consistent.

Instead, I think we can give the `strict` param to `search_builds_by_fields` (default to False), and only leave the `get_build_records_by_nvrs()` implementation which is more specific.

Reply:
sounds good to me